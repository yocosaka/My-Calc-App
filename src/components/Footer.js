import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => (
  <footer>
    <small>
      &copy;
      {' '}
      2021
      {' '}
      <Link to="https://github.com/yocosaka">Yoko Saka</Link>
    </small>
  </footer>
);

export default Footer;
