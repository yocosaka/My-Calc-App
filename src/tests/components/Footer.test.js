import React from 'react';
// import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Footer from '../../components/Footer';

// describe('Footer', () => {
//   test('renders Footer component ', () => {
//     render(<Footer />);
//     // screen.debug();
//     // eslint-disable-next-line no-unused-expressions
//     expect(screen.getByText('Yoko Saka')).toBeInTheDocument;
//   });
// });

describe('Test suits for Footer with Link', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
