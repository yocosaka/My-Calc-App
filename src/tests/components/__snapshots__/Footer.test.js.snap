// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Footer with Link should match with snapshot 1`] = `
<footer>
  <small>
    Â©
     
    2021
     
    <a
      href="/https://github.com/yocosaka"
      onClick={[Function]}
    >
      Yoko Saka
    </a>
  </small>
</footer>
`;
